bpo-29026: Clarify documentation of time.time

appeltel opened the pull request:
Clarify the documentation of time.time by more
precisely defining what is meant by "seconds since
the epoch" on most platforms. Additionally explain
how gmtime and localtime may be used to extract
calendar components and convert to a more common
date format.

@haypo 

https://bugs.python.org/issue29026

brettcannon commented:
Thanks for the PR, but https://github.com/python/cpython is a cpython mirror repository; pull requests are not accepted on this repo. Please submit bugs and patches at https://bugs.python.org.

appeltel committed:
bpo-29026: Clarity documentation of time.time

Clarify the documentation of time.time by more
precisely defining what is meant by "seconds since
the epoch" on most platforms. Additionally explain
how gmtime and localtime may be used to extract
calendar components and convert to a more common
date format.

vstinner commented:
Thank you for the new doc. It contains more info on timezone (UTC) and leap seconds (excluded).

vstinner commented on Doc/library/time.rst:
```diff
@@ -572,12 +579,27 @@ The module defines the following functions and data items:
+   On Windows and most Unix systems, the epoch is January 1, 1970,
+   00:00:00 (UTC) and leap seconds are not counted towards the time
+   in seconds since the epoch. This is commonly referred to as
+   `Unix time <https://en.wikipedia.org/wiki/Unix_time>`_.
```
IMHO it's worth it to repeat here:

To find out what the epoch is on a given platform, look at ``gmtime(0)``.

appeltel replied:
Repeated in d387bf7

vstinner commented on Doc/library/time.rst:
```diff
@@ -572,12 +579,27 @@ The module defines the following functions and data items:
 .. function:: time()
 
-   Return the time in seconds since the epoch as a floating point number.
+   Return the time in seconds since the epoch as a floating point
```
You should add a link on the "epoch" word to its definition.

appeltel replied:
Added link in d387bf7

vstinner reviewed (changes requested):
Sorry, I saw a few more minor issues.

berkerpeksag commented on Doc/library/time.rst:
```diff
@@ -19,9 +19,16 @@ An explanation of some terminology and conventions is in order.
-  1970.  To find out what the epoch is, look at ``gmtime(0)``.
+* The :dfn:`epoch` is the point where the time starts, and is platform
+  dependent.  For Unix, the epoch is January 1, 1970, 00:00:00 (UTC).
+  To find out what the epoch is on a given platform, look at ``gmtime(0)``.
```
I assume `gmtime(0)` is `time.gmtime(0)` here. If I'm correct, we can use this as an opportunity to make it clearer.

appeltel replied:
I changed it to `time.gmtime(0)` where it first appears in d387bf7. However, when the statement is repeated in the documentation of `time.time`, I retain `gmtime(0)` as this appears to be the convention used in all the function docs for the module.

appeltel replied:
Oops, failed to amend commit before pushing, I fixed to use doc conventions in 263a4f4

berkerpeksag commented on Doc/library/time.rst:
```diff
@@ -19,9 +19,16 @@ An explanation of some terminology and conventions is in order.
+
+* The term :dfn:`seconds since the epoch` refers to the total number
+  of elapsed seconds since the epoch, typically excluding
+  `leap seconds <https://en.wikipedia.org/wiki/Leap_second>`_.
```
I'd say add

```rst
.. `leap seconds`: https://en.wikipedia.org/wiki/Leap_second
```

and use 

```rst
`leap-seconds`_
```

everytime you need the Wikipedia article.

(I might the syntax got wrong so please try first :))

appeltel replied:
Added in d387bf7, also changed one other mention of leap seconds to link.

appeltel replied:
Whoops, failed to add to amended commit before push - added that other mention in 263a4f4

appeltel committed:
bpo-29026: Minor improvements for time.time doc

appeltel committed:
bpo-29026: Consistency fixes for time.time doc

vstinner commented:
Thanks! I merged the PR.
